storage/client/queries/queries.go

```diff
@@ -404,10 +404,29 @@ const (
 			($4::text IS NULL OR evs.tx_hash = $4::text OR evs.tx_eth_hash = $4::text) AND
 			($5::text IS NULL OR evs.type = $5::text) AND
 			($6::bytea IS NULL OR evs.evm_log_signature = $6::bytea) AND
-			($7::text IS NULL OR evs.related_accounts @> ARRAY[$7::text])
+			($7::text IS NULL OR evs.related_accounts @> ARRAY[$7::text]) AND
+			($8::text IS NULL OR (
+			    -- Currently this only supports EVM smart contracts.
+			    evs.type = 'evm.log' AND
+			    evs.body ->> 'address' = (
+			        SELECT encode(address_data, 'base64')
+			        FROM chain.address_preimages
+			        WHERE
+			            address = $8::text AND
+						context_identifier = 'oasis-runtime-sdk/address: secp256k1eth' AND
+						context_version = 0
+				)
+			)) AND
+			($9::text IS NULL OR (
+			    -- Currently this only supports ERC-721 Transfer events.
+			    evs.type = 'evm.log' AND
+			    evs.evm_log_signature = '\xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef' AND
```

Comment:
If you go with my push to require an explicit evm_log_signature param, this can go away because it will be enforced via $6.

Reply:
choosing not to require evm_log_signature param